src/Raven.Server/Web/RequestHandler.cs

```diff
@@ -159,6 +159,24 @@ private static bool HeadersAllowGzip(IDictionary<string, Microsoft.Extensions.Pr
             return false;
         }
 
+        protected void ValidateNodeForAddingToDb(string databaseName, string node, DatabaseRecord databaseRecord, ClusterTopology clusterTopology, string baseMessage = null)
```

Comment:
I would prefer this to be a static method somewhere. DatabaseHelper?

Reply:
Done